Issue running pytest locally for sct_propseg

running pytest locally, and I know that the folder sct_testing_data is present, i get this:

<details><summary>output</summary>

```
/Users/julien/code/sct/python/envs/venv_sct/bin/python "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/pydevd.py" --multiproc --qt-support=auto --client 127.0.0.1 --port 60779 --file "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py" --path /Users/julien/code/sct/unit_testing/cli/test_cli_sct_propseg.py
pydev debugger: process 28831 is connecting
Connected to pydev debugger (build 202.7660.27)
Launching pytest with arguments /Users/julien/code/sct/unit_testing/cli/test_cli_sct_propseg.py -s in /Users/julien/temp
Status: 100%|██████████| 7.17M/7.17M [00:00<00:00, 7.83MB/s]
Done!
============================= test session starts ==============================
platform darwin -- Python 3.6.12, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/julien/code/sct/python/envs/venv_sct/bin/python
cachedir: .pytest_cache
rootdir: /Users/julien/code/sct, configfile: setup.cfg
plugins: console-scripts-1.1.0, cov-2.10.1, forked-1.3.0
collecting ... collected 2 items
../code/sct/unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] >>> print(1)
1
# Running console script: sct_testing --function sct_propseg
# Script return code: 1
# Script stdout:
--
Spinal Cord Toolbox (git-jca/3044-ants-sampling-278f800ad8770e185d4864cb1bda92359284c67c*)
sct_testing --function sct_propseg
--
Creating temporary folder (/var/folders/s8/4qnm5q1n261ch35b5kkclsb00000gn/T/sct-20210113183622.244825-32q172nd)
Will run through the following tests:
- in parallel with 1 jobs: sct_propseg
Checking sct_propseg................................[FAIL]
   
   ====================================================================================================
   sct_propseg -i t2/t2.nii.gz -c t2 -qc testing-qc
   ====================================================================================================
   
   
   --
   Spinal Cord Toolbox (git-jca/3044-ants-sampling-278f800ad8770e185d4864cb1bda92359284c67c*)
   
   sct_propseg -i t2/t2.nii.gz -c t2 -qc testing-qc
   --
   
   Traceback (most recent call last):
   File "/Users/julien/code/sct/spinalcordtoolbox/scripts/sct_propseg.py", line 677, in <module>
   main(sys.argv[1:])
   File "/Users/julien/code/sct/spinalcordtoolbox/scripts/sct_propseg.py", line 664, in main
   img_seg = propseg(img_input, arguments)
   File "/Users/julien/code/sct/spinalcordtoolbox/scripts/sct_propseg.py", line 448, in propseg
   os.makedirs(folder_output)
   File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/os.py", line 220, in makedirs
   mkdir(name, mode)
   FileNotFoundError: [Errno 2] No such file or directory: ''
status: [1]
Failures: sct_propseg
Finished! Elapsed time: 1s
```

</details>

i'm trying to understand what's going on by debugging, but i get stuck there:
https://github.com/neuropoly/spinalcordtoolbox/blob/master/unit_testing/cli/test_cli_sct_propseg.py#L12

so, how are the input arguments "-i t2/t2.nii.gz -c t2 -qc testing-qc" passed on? where are they coming from?

Interesting. I can reproduce! (I wasn't expecting that, truthfully.)

<details>
<summary> Console output for <code>pytest unit_testing/cli/test_cli_sct_propseg.py</code> hidden, click to show. </summary>

```sh
(venv_sct) joshua@XPS-15-9560:~/repos/spinalcordtoolbox$ pytest unit_testing/cli/test_cli_sct_propseg.py 
Status: 100%|██████████████████████████████████████████████████| 7.17M/7.17M [00:00<00:00, 30.5MB/s]
Done!

======================================= test session starts ========================================
platform linux -- Python 3.6.12, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python
cachedir: .pytest_cache
rootdir: /home/joshua/repos/spinalcordtoolbox, configfile: setup.cfg
plugins: console-scripts-1.1.0, cov-2.10.1, forked-1.3.0, cases-3.1.1, xdist-2.2.0
[gw0] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox
[gw1] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox
[gw2] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox
[gw3] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox
[gw0] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
[gw1] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
[gw2] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
[gw3] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
Status: 100%|██████████████████████████████████████████████████| 7.17M/7.17M [00:00<00:00, 23.7MB/s]
Status: 100%|██████████████████████████████████████████████████| 7.17M/7.17M [00:00<00:00, 22.2MB/s]
Status: 100%|██████████████████████████████████████████████████| 7.17M/7.17M [00:00<00:00, 10.4MB/s]
Status: 100%|██████████████████████████████████████████████████| 7.17M/7.17M [00:00<00:00, 8.58MB/s]
gw0 [2] / gw1 [2] / gw2 [2] / gw3 [2]
scheduling tests via LoadScopeScheduling

unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] 
[gw0] [ 50%] FAILED unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] 
unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_o_flag 
[gw0] [100%] PASSED unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_o_flag 

============================================= FAILURES =============================================
__________________________ test_sct_propseg_backwards_compat[subprocess] ___________________________
[gw0] linux -- Python 3.6.12 /home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python

script_runner = <ScriptRunner subprocess>

    @pytest.mark.script_launch_mode('subprocess')
    def test_sct_propseg_backwards_compat(script_runner):
        ret = script_runner.run('sct_testing', '--function', 'sct_propseg')
        logger.debug(f"{ret.stdout}")
        logger.debug(f"{ret.stderr}")
>       assert ret.success
E       assert False
E        +  where False = <pytest_console_scripts.RunResult object at 0x7efc105dda20>.success

unit_testing/cli/test_cli_sct_propseg.py:15: AssertionError
--------------------------------------- Captured stdout call ---------------------------------------
# Running console script: sct_testing --function sct_propseg
# Script return code: 1
# Script stdout:

--
Spinal Cord Toolbox (git-master-2dc9c8a88449ce5ed4d0ae61caa9464e3da5ea6f)

sct_testing --function sct_propseg
--

Creating temporary folder (/tmp/sct-20210113221304.053082-_p6ffqz3)
Will run through the following tests:
- in parallel with 1 jobs: sct_propseg
Checking sct_propseg................................[FAIL]
   
   ====================================================================================================
   sct_propseg -i t2/t2.nii.gz -c t2 -qc testing-qc
   ====================================================================================================
   
   
   --
   Spinal Cord Toolbox (git-master-2dc9c8a88449ce5ed4d0ae61caa9464e3da5ea6f)
   
   sct_propseg -i t2/t2.nii.gz -c t2 -qc testing-qc
   --
   
   Traceback (most recent call last):
   File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py", line 677, in <module>
   main(sys.argv[1:])
   File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py", line 664, in main
   img_seg = propseg(img_input, arguments)
   File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py", line 448, in propseg
   os.makedirs(folder_output)
   File "/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/lib/python3.6/os.py", line 220, in makedirs
   mkdir(name, mode)
   FileNotFoundError: [Errno 2] No such file or directory: ''
status: [1]
Failures: sct_propseg
Finished! Elapsed time: 1s


# Script stderr:

========================================= warnings summary =========================================
unit_testing/cli/test_cli_sct_propseg.py: 14 warnings
  /home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/image.py:350: DeprecationWarning: get_data() is deprecated in favor of get_fdata(), which has a more predictable return type. To obtain get_data() behavior going forward, use numpy.asanyarray(img.dataobj).
  
  * deprecated from version: 3.0
  * Will raise <class 'nibabel.deprecator.ExpiredDeprecationError'> as of version: 5.0
    self.data = self.im_file.get_data()

unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_o_flag
  /home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_propseg.py:415: DeprecationWarning: get_data() is deprecated in favor of get_fdata(), which has a more predictable return type. To obtain get_data() behavior going forward, use numpy.asanyarray(img.dataobj).
  
  * deprecated from version: 3.0
  * Will raise <class 'nibabel.deprecator.ExpiredDeprecationError'> as of version: 5.0
    img_rescaled = nib.nifti1.Nifti1Image(img.get_data(), None, header=header_rescaled)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
===================================== short test summary info ======================================
FAILED unit_testing/cli/test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] - ...
============================= 1 failed, 1 passed, 15 warnings in 9.83s =============================
```

</details>

When I run `pytest unit_testing/` locally (i.e. the full test suite), I get the same result. This is the only test that fails for me.

The strange thing is, if I attach a debugger, the test passes?

<details>
<summary> Console output  for <code>pytest unit_testing/cli/test_cli_sct_propseg.py</code> with debugger attached, click to show </summary>

```sh
/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python /home/joshua/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.6682.179/plugins/python/helpers/pydev/pydevd.py --multiproc --qt-support=auto --client 127.0.0.1 --port 33995 --file /home/joshua/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.6682.179/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target test_cli_sct_propseg.py::test_sct_propseg_backwards_compat
Testing started at 10:15 p.m. ...
Connected to pydev debugger (build 203.6682.179)
Launching pytest with arguments test_cli_sct_propseg.py::test_sct_propseg_backwards_compat in /home/joshua/repos/spinalcordtoolbox/unit_testing/cli

Status: 100%|██████████| 7.17M/7.17M [00:00<00:00, 20.0MB/s]
Done!

============================= test session starts ==============================
platform linux -- Python 3.6.12, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python
cachedir: .pytest_cache
rootdir: /home/joshua/repos/spinalcordtoolbox, configfile: setup.cfg
plugins: console-scripts-1.1.0, cov-2.10.1, forked-1.3.0, cases-3.1.1, xdist-2.2.0
gw0 I / gw1 I / gw2 I / gw3 I
[gw0] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox/unit_testing/cli
[gw1] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox/unit_testing/cli
[gw2] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox/unit_testing/cli
[gw3] linux Python 3.6.12 cwd: /home/joshua/repos/spinalcordtoolbox/unit_testing/cli
[gw0] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
[gw1] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
[gw2] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
Status: 100%|██████████| 7.17M/7.17M [00:00<00:00, 25.8MB/s]
[gw3] Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56)  -- [GCC 7.3.0]
Status: 100%|██████████| 7.17M/7.17M [00:00<00:00, 29.8MB/s]
Status: 100%|██████████| 7.17M/7.17M [00:00<00:00, 30.0MB/s]
Status: 100%|██████████| 7.17M/7.17M [00:03<00:00, 2.34MB/s]
gw0 [1] / gw1 [1] / gw2 [1] / gw3 [1]

scheduling tests via LoadScopeScheduling

test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] 

============================== 1 passed in 19.86s ==============================

Process finished with exit code 0

[gw0] [100%] PASSED test_cli_sct_propseg.py::test_sct_propseg_backwards_compat[subprocess] 
```

Regardless of the intermittent behavior, I have a good idea of what change caused this...

--- a/spinalcordtoolbox/scripts/sct_propseg.py
+++ b/spinalcordtoolbox/scripts/sct_propseg.py
@@ -14,6 +14,7 @@
 # TODO: remove temp files in case rescaled is not "1"
 
 import os
+import pathlib
 import sys
 import logging
 
@@ -440,7 +441,7 @@ def propseg(img_input, options_dict):
     else:
         fname_out = os.path.basename(add_suffix(fname_data, "_seg"))
     
-    folder_output = os.path.dirname(fname_out)
+    folder_output = str(pathlib.Path(fname_out).parent)
     cmd += ['-o', folder_output]
     if not os.path.isdir(folder_output) and os.path.exists(folder_output):
         logger.error("output directory %s is not a valid directory" % folder_output)

